### PR TITLE
Add LocationLists and RangeLists

### PR DESCRIPTION
**Pull request by philipc**
These allow users to parse lists without caring about DWARF version details.

The old debug_loc/debug_ranges interfaces are removed, since LocationsLists/RangeLists parse these now, and it was too easy to wrongly use them for DWARF version 5.

cc @rocallahan

### CODE REVIEW

File: src/loclists.rs [hunk 1]
```diff
@@ -39,39 +39,83 @@ where
     }
 }
 
+/// An offset into either the `.debug_loc` section or the `.debug_loclists` section,
+/// depending on the version of the unit the offset was contained in.
```
Comment (fitzgen) [inline src/loclists.rs]: Should we return an `enum` of either a `DebugLocOffset` or a `DebugLocListsOffset`? I'm concerned that we are making it easier to misuse our APIs and weakening the benefits that strongly typing offsets gives.

Comment (fitzgen) [reply src/loclists.rs]: Nevermind -- I understand how this works better now.

**Comment by fitzgen** [comment]
Thanks @philipc !

File: src/loclists.rs (hunk 1 above)
Comment (philipc) [reply src/loclists.rs]: Yeah I wasn't sure which way was better.  We could make it an enum, and then you wouldn't need to pass the unit version into the other functions, but you would still need to pass the unit address size (or we could include that in the enum too).

**Pull request merged**